src/mono/wasm/debugger/BrowserDebugProxy/EvaluateExpression.cs

```diff
@@ -277,8 +277,20 @@ public static string ConvertJSToCSharpLocalVariableAssignment(string idName, JTo
                         switch (objectId?.Scheme)
                         {
                             case "valuetype" when variable["isEnum"]?.Value<bool>() == true:
-                                typeRet = variable["className"]?.Value<string>();
-                                valueRet = $"({typeRet}) {value["value"].Value<double>()}";
+                                 // using Enum Type from variable["className"] requires adding references to assemblies
+                                 // with this type definition - it's easier to use the underlying numeric type
+                                typeRet = "double";
+                                // when indexing with enums, enum object is packed in another enum
+                                // which is not the case in other scenarios
+                                try
+                                {
+                                    valueRet = $"({typeRet}) {value["value"].Value<double>()}";
+                                }
+                                catch (InvalidCastException)
+                                {
+                                    // value["value"] is JObject with another "value" field
+                                    valueRet = $"({typeRet}) {value["value"]["value"].Value<double>()}";
```

Comment:
Should we check if this also throws an exception?

Reply:
I will add it, it might be easier to read the logs later. I also checked all the usages of this function and they are secured by try-catch on higher levels so it would have gotten caught.